### PR TITLE
Pagination support for '/current'

### PR DESCRIPTION
**Pull request by om-sharma**
Response for `/current` has LinkHeader for next and previous page.  Also html representation renders the Next and Previous page links

### CODE REVIEW

File: src/main/java/uk/gov/register/presentation/dao/RecentEntryIndexQueryDAO.java [hunk 1]
```diff
@@ -31,9 +31,12 @@
...
-    @SqlQuery("SELECT SERIAL_NUMBER,ENTRY FROM ORDERED_ENTRY_INDEX WHERE SERIAL_NUMBER IN(SELECT SERIAL_NUMBER FROM CURRENT_KEYS ORDER BY SERIAL_NUMBER DESC LIMIT :limit) ORDER BY SERIAL_NUMBER DESC")
-    List<DbEntry> getLatestEntriesOfRecords(@Bind("limit") long maxNumberToFetch);
+    @SqlQuery("SELECT SERIAL_NUMBER,ENTRY FROM ORDERED_ENTRY_INDEX WHERE SERIAL_NUMBER IN(SELECT SERIAL_NUMBER FROM CURRENT_KEYS ORDER BY SERIAL_NUMBER DESC LIMIT :limit OFFSET :offset) ORDER BY SERIAL_NUMBER DESC")
+    List<DbEntry> getLatestEntriesOfRecords(@Bind("limit") long maxNumberToFetch, @Bind("offset") long offset);
 
     @SqlQuery("SELECT COUNT FROM REGISTER_ENTRIES_COUNT")
-    int getTotalEntriesCount();
+    int getTotalEntries();
+
+    @SqlQuery("SELECT COUNT(*) FROM CURRENT_KEYS")
```
Comment (philandstuff) [inline src/main/java/uk/gov/register/presentation/dao/RecentEntryIndexQueryDAO.java:40]: does this get slow for large registers?

Comment (om-sharma) [inline src/main/java/uk/gov/register/presentation/dao/RecentEntryIndexQueryDAO.java:40]: I didn't check this yet, was waiting for data load in address register to complete. 

I ran this query with 10,000,000 entries in the table and it wasn't very bad. Will check it again once data loading is done.

Comment (om-sharma) [inline src/main/java/uk/gov/register/presentation/dao/RecentEntryIndexQueryDAO.java:40]: so count(*) on current_keys table with 35,009,563 entries takes 25 seconds. Which means that we need to browse other solutions. I have got some other workarounds, let's discuss it tomorrow.

**Comment by om-sharma** [comment]
Discussed IRL, This work was required for country widget. we decided to use /feed call and normalize data in widget. will reopen again when we need pagination for '/current' again.

**Pull request closed without merging**